packages/dev/core/src/Materials/GreasedLine/greasedLinePluginMaterial.ts

```diff
@@ -458,7 +460,14 @@ export class GreasedLinePluginMaterial extends MaterialPluginBase implements IGr
                             #ifdef GREASED_LINE_COLOR_DISTRIBUTION_TYPE_LINE
                                 vec4 grlColor = texture2D(grl_colors, vec2(grlCounters, 0.), 0.);
                             #else
-                                vec4 grlColor = texture2D(grl_colors, vec2(grlColorPointer/grlColorsWidth, 0.), 0.);
+                                vec2 lookup;
+                                if (grl_textureSize.y <= 1.0) {
+                                    lookup = vec2(fract(grlColorPointer / grl_textureSize.x), 0.0);
+                                } else {
+                                    lookup = vec2(fract(grlColorPointer / grl_textureSize.x),
+                                                  1. - floor(grlColorPointer / grl_textureSize.x) / (grl_textureSize.y - 1.0));
+                                }
+                                vec4 grlColor = texture2D(grl_colors, lookup, 0.0);

```

Comment:
You can avoid the if/else construct and simplify the code with something like:
```glsl
lookup.y = floor(grlColorPointer / grl_textureSize.x);
lookup.x = grlColorPointer % grl_textureSize.x;
```
With this change, you now only need `grl_textureSize.x`, so you can only pass a float, which will align better in the ubo struct, as the previous data (`grl_singleColor`) is a vec3.

Reply:
I think `lookup.x` and `lookup.y` should be bounded between 0 and 1, and y is flipped, that is why there is a `1. - floor(grlColorPointer / grl_textureSize.x) / (grl_textureSize.y - 1.0)`. In later versions of glsl, you can get the sampler size with a buildin function.
I may be able to move the calculation for the texture size to the shader, but there will likely still be a comparison between texture size (as a float) and the grlColorPointer, but I am not sure if it is a good idea ?

Reply:
The alternative may be to:

remove the if-else and:
```
vec2 lookup = vec2(
    fract(grlColorPointer / grl_textureSize.x),
    1.0 - floor(grlColorPointer / grl_textureSize.x) / max(grl_textureSize.y - 1.0, 1.0)
);
```

Reply:
`grl_textureSize.y` is not the same as the max texture size that can be handled by the engine, but rather the actual texture height that is sampled from (likely with a filler if texture height is > 1, otherwise if height = 1, no filler is needed).

Reply:
I'm fine with your latest iteration!